File: FinalBootMVC/src/main/java/com/springboot/pkg/model/Adresse.java
package com.springboot.pkg.model;

import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;
import com.springboot.pkg.model.jsonview.JsonViews;


@Embeddable
public class Adresse {

	@JsonView(JsonViews.Common.class)
	private Integer num;
	@JsonView(JsonViews.Common.class)
	private Integer cp;
	@JsonView(JsonViews.Common.class)
	private String rue;
	@JsonView(JsonViews.Common.class)
	private String ville;
	@JsonView(JsonViews.Common.class)
	private String pays;
	
	
	public Adresse() {
		
	}


	public Adresse(Integer num, Integer cp, String rue, String ville, String pays) {
	
		this.num = num;
		this.cp = cp;
		this.rue = rue;
		this.ville = ville;
		this.pays = pays;
	}


	public Integer getNum() {
		return num;
	}


	public void setNum(Integer num) {
		this.num = num;
	}


	public Integer getCp() {
		return cp;
	}


	public void setCp(Integer cp) {
		this.cp = cp;
	}


	public String getRue() {
		return rue;
	}


	public void setRue(String rue) {
		this.rue = rue;
	}


	public String getVille() {
		return ville;
	}


	public void setVille(String ville) {
		this.ville = ville;
	}


	public String getPays() {
		return pays;
	}


	public void setPays(String pays) {
		this.pays = pays;
	}
	
	
}

File: FinalAngular/src/app/model/adresse.ts
export class Adresse {
  constructor(private _num: number, private _cp: number, private _rue: string,
    private _ville?: string, private _pays?: string) {

  }

  public get rue(): string {
    return this._rue;
  }
  public set rue(_rue: string) {
    this._rue = _rue;
  }

  public get ville(): string {
    return this._ville;
  }
  public set ville(_ville: string) {
    this._ville = _ville;
  }
  public get pays(): string {
    return this._pays;
  }
  public set pays(_pays: string) {
    this._pays = _pays;
  }

  public get num(): number {
    return this._num;
  }
  public set num(_num: number) {
    this._num = _num;
  }
  public get cp(): number {
    return this._cp;
  }
  public set cp(_cp: number) {
    this._cp = _cp;
  }
}
